src/main/java/teammates/ui/webapi/FeedbackSessionLogsUpdateAction.java

```diff
@@ -0,0 +1,62 @@
+package teammates.ui.webapi;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import teammates.common.datatransfer.attributes.FeedbackSessionLogEntryAttributes;
+import teammates.common.exception.InvalidParametersException;
+import teammates.common.util.Logger;
+
+/**
+ * Action: Sync feedback session logs from GCloud logging service.
+ */
+public class FeedbackSessionLogsUpdateAction extends AdminOnlyAction {
+
+    private static final Logger log = Logger.getLogger();
+
+    @Override
+    public ActionResult execute() {
+        Instant currentTime = Instant.now();
+        Map<String, Map<String, Long>> studentLatestLogs = new HashMap<>();
+        List<FeedbackSessionLogEntryAttributes> validLogEntries = new ArrayList<>();
+        List<FeedbackSessionLogEntryAttributes> allLogEntries =
+                logsProcessor.getFeedbackSessionLogs(null, null,
+                        currentTime.minus(15, ChronoUnit.MINUTES).toEpochMilli(), currentTime.toEpochMilli(), null);
+
+        for (FeedbackSessionLogEntryAttributes logEntry : allLogEntries) {
+            String studentEmail = logEntry.getStudentEmail();
+            String logType = logEntry.getFeedbackSessionLogType();
+            Long logTimestamp = logEntry.getTimestamp();
+            Map<String, Long> studentLog = studentLatestLogs.getOrDefault(studentEmail, new HashMap<>());
+            boolean isValid = true;
+
+            if (studentLog.containsKey(logType)) {
+                Long lastStudentLogTimestamp = studentLog.get(logType);
+
+                if (Math.abs(lastStudentLogTimestamp - logTimestamp) < 2 * 1000) {
+                    isValid = false;
```

Comment:
Logs with close timestamps should not be duplicated, but we still want to know the number of occurrences (e.g. within 2 seconds, student A tries to submit 4 times).

Reply:
You need to keep either the first or the last log and also store the info that there are 4 such attempts in the 2 seconds span.

Reply:
> also store the info that there are 4 such attempts in the 2 seconds span.

How do we store this info?

Reply:
I can give an answer, but I'll be doing your work if I do.

Reply:
@wkurniawan07 do the 4 logs have to be of 1 type for them to be invalid? I was only considering close logs of the same type to be invalid, as @damithc instructed

Reply:
Yes, my view is that rapid repeating of the same action is a 'possible spam' (best not to call them 'invalid'). If the user clicks submission links of two sessions rapidly, that is unlikely a spam. e.g.. the user could be opening links received in a 'links recovery' email.